### PR TITLE
feat: add security scan

### PR DESCRIPTION
**Pull request by nicolascb**
# Description

Implement a security scan with [trivy](https://github.com/aquasecurity/trivy).

## Readiness checklist

* [ ] I added tests for new functionality or bugfixes.
* [ ] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [ ] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [ ] I marked all done items in this checklist.

### CODE REVIEW

File: Taskfile.yml [hunk 1]
```diff
@@ -340,3 +340,20 @@ tasks:
...
+
+  security-scan-fs:
+    cmds:
+      - docker run --rm -v $(pwd):/app -w /app aquasec/trivy:latest fs . -q --exit-code=1
+
+  security-scan-docker:
+    cmds:
+      - docker build -q -t ferretdb/ferretdb:trivy-scan .
+      - docker save ferretdb/ferretdb:trivy-scan > docker-image.tar
+      - docker run --rm -v $(pwd):/app -w /app aquasec/trivy:latest image --input docker-image.tar -q --exit-code=1
```
Comment (AlekSi) [inline Taskfile.yml:352]: I think that check does not produce anything useful for us, let's remove it

File: Taskfile.yml [hunk 2]
```diff
@@ -340,3 +340,20 @@ tasks:
         sh: cat ../internal/util/version/gen/version.txt
       GOARCH:
         sh: go env GOARCH
+
+  security-scan-fs:
+    cmds:
+      - docker run --rm -v $(pwd):/app -w /app aquasec/trivy:latest fs . -q --exit-code=1
```
Comment (AlekSi) [inline Taskfile.yml:346]: That one might be useful, let's keep it. But let's use the same dependency mechanism we use for other Docker images; see https://github.com/FerretDB/FerretDB/tree/main/build/deps

**Comment by nicolascb** [comment]
hello @AlekSi , please, review again?

File: .gitignore [hunk 3]
```diff
@@ -25,3 +25,5 @@ testdata/fuzz/
 
 # for local Docker Compse changes; see https://docs.docker.com/compose/extends/#multiple-compose-files
 docker-compose.override.yml
+
+docker-image.tar
```
Comment (AlekSi) [inline .gitignore:29]: That's no longer needed

File: .github/workflows/security.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup Go
+        uses: FerretDB/github-actions/setup-go@main
+        with:
+          cache-key: test
```
Comment (AlekSi) [inline .github/workflows/security.yml:36]: We should use an unique cache key. That should be "security-scan" or something

**Review by AlekSi — COMMENTED** [review]
Let's do a few tweaks

File: .github/workflows/security.yml (hunk 4 above)
Comment (AlekSi) [reply .github/workflows/security.yml:36]: Actually, I think that's fine :)

**Comment by AlekSi** [comment]
I could not push my changes to this PR, so I created a new one: #1142

**Pull request closed without merging**